src/v/transform/rpc/client.cc

```diff
@@ -807,6 +808,42 @@ client::generate_remote_report(model::node_id node) {
     co_return std::move(resp).value().report;
 }
 
+std::optional<cluster::topic_configuration>
+client::find_topic_cfg(model::topic_namespace_view tp_ns) const {
+    return _topic_metadata->find_topic_cfg(tp_ns);
+}
+
+ss::future<cluster::errc> client::create_transform_logs_topic() {
+    co_return co_await retry(
+      [this]() { return try_create_transform_logs_topic(); });
+}
+
+ss::future<cluster::errc> client::try_create_transform_logs_topic() {
+    cluster::topic_properties topic_props;
+    topic_props.batch_max_bytes
+      = config::shard_local_cfg().kafka_batch_max_bytes();
+    topic_props.retention_bytes = tristate<size_t>(-1);
+    topic_props.retention_duration = tristate<std::chrono::milliseconds>(
+      std::chrono::duration_cast<std::chrono::milliseconds>(24h));
+    topic_props.cleanup_policy_bitflags
+      = model::cleanup_policy_bitflags::deletion;
+    auto fut = co_await ss::coroutine::as_future<cluster::errc>(
+      _topic_creator->create_topic(
+        model::topic_namespace_view(model::transform_log_internal_nt),
+        config::shard_local_cfg().default_topic_partitions(),
+        std::move(topic_props)));
+    if (fut.failed()) {
+        throw std::runtime_error("Error creating transform logs topic");
```

Comment:
this will result in an "Exceptional future ignored" error. When you have a failed future in hand, you need to eat the exception out of it. So you can do `fmt::format("{}", fut.get_exception());` or whatever else, including ignore_ready_future if you really don't care about the failure.